addons/website_hr_recruitment/controllers/main.py

```diff
@@ -80,6 +80,18 @@ def jobs_add(self, **kwargs):
         })
         return request.redirect(request.env["website"].get_client_action_url(f"/jobs/detail/{slug(job)}", True))
 
+    @http.route('/jobs/duplicate', type='http', auth="user", methods=['POST'], website=True)
+    def jobs_duplicate(self, job_id, **post):
+        job = request.env['hr.job'].browse(int(job_id))
```

Comment:
Maybe the route could have been changed to include a model converter, and the loop can probably be improved but didn't check further, I think this should be removed and made generic (see other comment on JS part)